diff --git a/node_modules/react-native-calendars/src/expandableCalendar/index.js b/node_modules/react-native-calendars/src/expandableCalendar/index.js
index 4fb12b0..4ffb9fd 100644
--- a/node_modules/react-native-calendars/src/expandableCalendar/index.js
+++ b/node_modules/react-native-calendars/src/expandableCalendar/index.js
@@ -280,9 +280,10 @@ const ExpandableCalendar = (props) => {
                 speed: SPEED,
                 bounciness: BOUNCINESS,
                 useNativeDriver: false
-            }).start();
+            }).start(() => {
+                setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
+            });;
             onCalendarToggled?.(_isOpen);
-            setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
             closeHeader(_isOpen);
             resetWeekCalendarOpacity(_isOpen);
         }
